### PR TITLE
improve the float warning, add flag for treatment of floats

### PR DESCRIPTION
**Pull request by cbm755**

### CODE REVIEW

**Comment by cbm755** [comment]
This adds `sym (0.1, 'f')` support which gives the exact rational corresponding to a float.

`sym(inf, 'f')` and `sym(nan, 'f')` are broken: I added xtests.  Easy enough to fix but I'll do it as part of a larger refactoring in another PR.  

@mtmiller could you proofread these changes to the docs and the float warning?

@latot this shouldn't change the default behaviour: mostly indenting changes and re-ordering things a little.

**Comment by latot** [comment]
:)

File: inst/@sym/sym.m [hunk 1]
```diff
@@ -64,31 +67,60 @@
...
+%% @end group
+%% @end example
+%%
+%% The ``danger'' here is that typing @code{0.1} gives a double-precision
+%% floating-point value which differs slightly from the fraction
+%% @code{sym(1)/10} (and this is true for most decimal expressions).
+%% It is generally impossible to determine which exact symbolic value the
+%% user intended.
+%% The warning indicates that some heuristics have been applied,
+%% namely a preferance for ``small'' fractions (and small fractions
```
Comment (mtmiller) [inline inst/@sym/sym.m]: preference

File: inst/@sym/sym.m [hunk 2]
```diff
@@ -64,31 +67,60 @@
...
-%%   @result{} (sym) False
+%% [sym(pi) sym(3)/2; sym(1) 0]
+%%   @result{} (sym 2×2 matrix)
+%%       ⎡π  3/2⎤
+%%       ⎢      ⎥
+%%       ⎣1   0 ⎦
+%% @end group
+%% @end example
+%% (Careful: at least one entry per row must be @code{sym} to workaround
+%% a GNU Octave bug @url{https://savannah.gnu.org/bugs/index.php?42152}.)
```
Comment (mtmiller) [inline inst/@sym/sym.m]: I prefer the shorter form `https://savannah.gnu.org/bugs/?42152`

**Comment by cbm755** [comment]
Thank you both, merging

**Pull request merged**